cimi-resources/test/com/sixsq/slipstream/ssclj/resources/external_object_lifecycle_test_utils.clj

```diff
@@ -108,361 +118,324 @@
+                             :request-method :post)
+                    (ltu/body->edn)
+                    (ltu/is-status 403))
+
+                (-> session-user-no-view
+                    (request abs-upload-uri
+                             :request-method :post)
+                    (ltu/body->edn)
+                    (ltu/is-status 403))
+
+                (-> session-user-view
+                    (request abs-upload-uri
+                             :request-method :post)
+                    (ltu/body->edn)
+                    (ltu/is-status 403))
+
+                ;; owner can trigger the upload action
+                (-> session
+                    (request abs-upload-uri
+                             :request-method :post)
+                    (ltu/body->edn)
+                    (ltu/is-status 200))
+
+                ;; after getting upload URL the state is set to 'uploading'
+                ;; 'ready', 'upload', and 'delete' actions are present
+                (-> session
+                    (request abs-uri)
+                    (ltu/body->edn)
+                    (ltu/is-status 200)
+                    (ltu/is-key-value :state eo/state-uploading)
+                    (ltu/is-operation-present "ready")
+                    (ltu/is-operation-present "delete")
+                    (ltu/is-operation-present "edit")
+                    (ltu/is-operation-present "upload")
+                    (ltu/is-operation-absent "download"))
+
+                ;; user with view access should see change of state
+                ;; actions should be the same
+                (-> session-user-view
+                    (request abs-uri)
+                    (ltu/body->edn)
+                    (ltu/is-status 200)
+                    (ltu/is-key-value :state eo/state-uploading)
+                    (ltu/is-operation-absent "ready")
+                    (ltu/is-operation-absent "delete")
+                    (ltu/is-operation-absent "edit")
+                    (ltu/is-operation-absent "upload")
+                    (ltu/is-operation-absent "download"))
+
+                ;; doing it again should succeed, a new upload URL can be obtained
+                ;; in 'uploading' state
+                (-> session
+                    (request abs-upload-uri
+                             :request-method :post)
+                    (ltu/body->edn)
+                    (ltu/is-status 200))
+
+                (let [uploading-eo (-> session
+                                       (request abs-uri)
+                                       (ltu/body->edn)
+                                       (ltu/is-operation-present "ready")
+                                       (ltu/is-status 200))
+
+                      ready-url-action (str p/service-context (ltu/get-op uploading-eo "ready"))]
+
+
+                  ;; triggering the ready url with anonymous, authorized or unauthorized viewer should fail
+                  (-> session-anon
```

Comment:
Same as above : The 3 requests could be grouped in a `doseq`

Reply:
Done.